florist/api/models/common.py

```diff
@@ -0,0 +1,49 @@
+"""Common classes and functions for models."""
+
+import json
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from torch import nn
+
+
+class AbstractModel(nn.Module, ABC):
+    """Common functions and abstract methods for models."""
+
+    @classmethod
+    @abstractmethod
+    def mandatory_server_info_fields(cls) -> List[str]:
+        """
+        Return a list of server info fields that are mandatory by this model.
+
+        Must be implemented by the subclasses.
+
+        :return: A list of fields that must be present in the server info for this model.
+        """
+        pass
+
+    @classmethod
+    def parse_server_info(cls, server_info: str) -> Dict[str, Any]:
```

Comment:
Similar comments as below, this seems to be outside what I would expect a model class to handle. If you think different, happy to hear your thoughts :)

Reply:
Let's discuss it on the other comment.